src/service.rs

```diff
@@ -265,41 +290,67 @@ impl DaService for BitcoinService {
 
         let blob = blob.to_vec();
         let network = self.network;
-        let address = self.address.clone();
+        let address = self
+            .address
+            .clone()
+            .require_network(network)
+            .expect("Invalid network for address");
```

Comment:
This should be validated in the with_client not here

Reply:
see reply I left on the first comment